plugins/usbpro/DMXUSBDevice.cpp

```diff
@@ -0,0 +1,269 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * DMXUSBDevice.cpp
+ * A DMXUSB Device (based on DMX King Ultra DMX Pro code by Simon Newton)
+ * Copyright (C) 2019 Perry Naseck (DaAwesomeP)
+ *
+ * This device creates n ports, 0 input and n outputs (fetchs number from
+ * device).
+ */
+
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+#include "common/rpc/RpcController.h"
+#include "ola/Callback.h"
+#include "ola/Constants.h"
+#include "plugins/usbpro/DMXUSBDevice.h"
+#include "plugins/usbpro/UsbProWidgetDetector.h"
+
+namespace ola {
+namespace plugin {
+namespace usbpro {
+
+using ola::plugin::usbpro::Request;
+using ola::plugin::usbpro::Reply;
+using ola::rpc::RpcController;
+using std::string;
+
+/*
+ * Create a new device
+ * @param owner  the plugin that owns this device
+ * @param name  the device name
+ * @param dev_path  path to the pro widget
+ */
+DMXUSBDevice::DMXUSBDevice(ola::PluginAdaptor *plugin_adaptor,
+                                     ola::AbstractPlugin *owner,
+                                     const string &name,
+                                     DMXUSBWidget *widget,
+                                     OLA_UNUSED uint16_t esta_id,
+                                     OLA_UNUSED uint16_t device_id,
+                                     uint32_t serial,
+                                     uint16_t firmware_version,
+                                     unsigned int fps_limit):
+    UsbSerialDevice(owner, name, widget),
+    m_plugin_adaptor(plugin_adaptor),
+    m_dmxusb_widget(widget),
+    m_serial(),
+    m_got_parameters(false),
+    m_got_extended_parameters(false),
+    m_break_time(0),
+    m_mab_time(0),
+    m_rate(0),
+    m_out_ports(0),
+    m_in_ports(0),
```

Comment:
Isn't this always zero?

Reply:
I haven't implemented input ports yet on the Arduino side, so this is really a placeholder until I do.

Reply:
It's probably better to comment out, or ideally remove the lines, rather than having them there without the functionality. It didn't look like the rest of the C++ code was present either from what I could see. If it was all in one commit you could always revert it in future when you have.